### PR TITLE
Add hidden nodes for hourglass

### PR DESCRIPTION
**Pull request by rtclay**
Use hidden nodes to fill the space where missing ancestors would go on the chart

### CODE REVIEW

**Comment by rtclay** [comment]
hourglass001

Without:
hourglass002
With:
hourglass003

**Comment by prculley** [comment]
You charts are so large with low resolution I cannot really see what is different, but it appears that some of the parents in the 'with' chart are missing; is it possible you have the images swapped, or do I not understand what you are doing?

**Comment by prculley** [comment]
After reading the code I understand now; I think your tooltip at least should have some additional explanation:
"This option causes space to reserved on the chart where missing ancestors might eventually be found."
Most users son't have any idea what 'nodes' means, hidden or otherwise.

**Comment by rtclay** [comment]
Good idea. I'll try to make clearer.

File: gramps/plugins/graph/gvhourglass.py [hunk 1]
```diff
@@ -158,22 +162,32 @@ def traverse_down(self, person, gen):
...
 
     def traverse_up(self, person, gen):
         """
         Recursively find the ancestors of the given person.
         """
         if gen > self.max_ascend:
             return
-        family_handle = person.get_main_parents_family_handle()
+
+        try:
```
Comment (Nick-Hall) [inline gramps/plugins/graph/gvhourglass.py]: The following might be slightly clearer:
```
if isinstance(person, int):
    person_id = person
    family_handle = None
else:
    person_id = person.get_gramps_id()
    family_handle = person.get_main_parents_family_handle()
```

**Comment by Nick-Hall** [comment]
Rebased.

**Comment by Nick-Hall** [comment]
@rtclay Is this ready to merge?

**Comment by Nick-Hall** [comment]
After merging #671 this PR now has conflicts.  Unfortunately, I don't know the code well enough to resolve them.

**Comment by Nick-Hall** [comment]
I am closing this now.  Please feel free to resubmit if you can fix the conflicts.

**Pull request closed without merging**